### DISCUSSION TITLE
About licence of plugin

### DISCUSSION THREAD

**Discussion by simonetravaglini** [nocodb/nocodb#5576] [Q&A]
Hi team, you are doing a great work.
One question,
if I deploy an instance of nocodb for a customer of mine, and develop a plugin, plugin should be release with agpl3 licence or can be released also with commercial license?
Thanks for support.

**Comment by wingkwong** [comment]
If you modify our source code, you need to give the public access to the source code of the modified version.

**Reply by simonetravaglini** [reply]
@wingkwong thanks for answer.
So this is true also for plugin and not only for core files?

**Reply by o1lab** [reply]
If your plugin uses nocodb files - then yes. If not this could be standalone. However, as Wing all changes to nocodb has to be made open source. If you don't mind sharing what is the plugin about. If its not really propreitary - it could be useful for community.
